Stellarium/stellarium
# Ignore autogen files

Ignore automatically generated files.

### Description



- ignore `__` directories
- ignore `cmake` auto generated files
- ignore `/translations` directory because it will be auto-generated when `cmake`
- ignore `doxygen` auto generated files
- ignore `doc/stellarium.pod` and `doc/stellarium.1`
- ignore auto generated `.h` files under `src/external`
- ignore all `Makefile` but keep `guide/Makefile`
- ignore `ui_*.h` files generated by Qt from `*.ui` files
- ignore `qrc_*.cpp` generated by Qt from `*.qrc` files
- ignore `*_autogen` directories
- ignore auto generated `*.a` static libraries

Fixes # (issue)

### Screenshots (if appropriate):

### Type of change


### How Has This Been Tested?



1. Run `cmake .` and `make` to do a full build to see which files are introduced.
1. Prepare a new pattern which matches a bunch of filenames we want to ignore.
1. Run `git clean -fx .` to clean all of the auto generated files.
1. Search the pattern mentioned in step 2 in project root to see if there is any version controlled paths matching the pattern.
1. No result in step 3 means this is a good pattern. Add it into `.gitignore`. Else, try to narrow the pattern down.


**Test Configuration**:
* Operating system: **not applicable to this PR**
* Graphics Card: **not applicable to this PR**

## Checklist:
